cmake/pcl_find_avx.cmake

```diff
@@ -0,0 +1,40 @@
+###############################################################################
+# Check for the presence of AVX and figure out the flags to use for it.
+macro(PCL_CHECK_FOR_AVX)
+    set(AVX_FLAGS)
+    set(AVX_DEFINITIONS)
+
+    include(CheckCXXSourceRuns)
+
+    check_cxx_source_runs("
+        #include <immintrin.h>
+        int main()
+        {
+             __m256 a;
+             a = _mm256_set1_ps(0);
+             return 0;
+        }"
+        HAVE_AVX)
+    
+    check_cxx_source_runs("    
+          #include <immintrin.h>
+        int main()
+        {
+            __m256i a = {0};
+            a = _mm256_abs_epi16(a);
+            return 0;
+        }"
+        HAVE_AVX2)
+        
+    if(MSVC)
+        if(HAVE_AVX)
+            set(AVX_DEFINITIONS "${AVX_DEFINITIONS} -D__AVX__")
```

Comment:
```suggestion
            string(APPEND AVX_DEFINITIONS " -D__AVX__")
```

Reply:
As you already know it is MSVC - is there anything that prevent you from calling `add_definitions` directly without extra variable?

Reply:
Probably not. It was only for keeping the same structure as with SSE. And if it make sense to add cases for Unix this is also the way to go? Or can `add_definitions` be used either way?
Then we can probably change it.

Reply:
Since the file is called `pcl_find_avx `- it should probably return two variables, ie. `AVX_DEFINITIONS `and `AVX_FLAGS`. I'll change that and aftwards use them in the main CMakeLists.